src/point-cloud-layer.ts

```diff
@@ -0,0 +1,209 @@
+import {
+  CompositeLayer,
+  CompositeLayerProps,
+  DefaultProps,
+  GetPickingInfoParams,
+  Layer,
+  LayersList,
+  assert,
+  Unit,
+  Material,
+} from "@deck.gl/core/typed";
+import { PointCloudLayer } from "@deck.gl/layers/typed";
+import type { PointCloudLayerProps } from "@deck.gl/layers/typed";
+import * as arrow from "apache-arrow";
+import * as ga from "@geoarrow/geoarrow-js";
+import {
+  assignAccessor,
+  extractAccessorsFromProps,
+  getGeometryVector,
+} from "./utils.js";
+// TODO which accessors are actually needed for a pointcloud layer
+import {
+  GeoArrowExtraPickingProps,
+  computeChunkOffsets,
+  getPickingInfo,
+} from "./picking.js";
+import { ColorAccessor, GeoArrowPickingInfo, NormalAccessor } from "./types.js";
+import { EXTENSION_NAME } from "./constants.js";
+import { validateAccessors } from "./validate.js";
+
+/* All properties supported by GeoArrowPointCloudLayer */
+export type GeoArrowPointCloudLayerProps = Omit<
+  PointCloudLayerProps<arrow.Table>, // TODO is this still an Arrow Table or is it another datatype? An arrow table is a vector/array like value so is it the same?
```

Comment:
It's still an Arrow Table, because we can structure the data as attributes that are associated with each point.